src/types/utility-types/enumeration.ts

```diff
@@ -0,0 +1,29 @@
+import { ISimpleType } from "../type"
+import { union } from "./union"
+import { literal } from "./literal"
+
+export function enumeration(options: string[]): ISimpleType<string>
+export function enumeration(name: string, options: string[]): ISimpleType<string>
+/**
+ * Can be used to create an string based enumeration.
+ * (note: this methods is just sugar for a union of string literals)
+ *
+ * @example
+ * ```javascript
+ * const TrafficLight = types.model({
+ *   color: types.enum("Color", ["Red", "Orange", "Green"])
+ * })
+ * ```
+ *
+ * @export
+ * @alias types.enumeration
+ * @param {string} name descriptive name of the enumeration (optional)
+ * @param {string[]} options possible values this enumeration can have
+ * @returns {ISimpleType<string>}
+ */
+export function enumeration(name: string | string[], options?: string[]): ISimpleType<string> {
```

Comment:
could be <T> - T[] - ISimpleType<T>
or if only strings and numbers are supported then string | number